Sources/ArcGISToolkit/FloorFilter/FloorFilterViewController.swift

```diff
@@ -0,0 +1,534 @@
+            floorFilterStackView.setNeedsLayout()
+        }
+    }
+}
+
+extension FloorFilterViewController: UITableViewDataSource, UITableViewDelegate {
+    
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return state == .partiallyExpanded ? 1 : viewModel.visibleLevelsInExpandedList.count
+    }
+    
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "levelCell", for: indexPath)
+        let levels = viewModel.visibleLevelsInExpandedList
+        
+        // Style the cell
+        cell.heightAnchor.constraint(equalToConstant: CGFloat(buttonHeight)).isActive = true
+        cell.widthAnchor.constraint(equalToConstant: CGFloat(buttonWidth)).isActive = true
+        cell.textLabel?.font = UIFont(name: fontName, size: fontSize)
+        cell.textLabel?.adjustsFontSizeToFitWidth = true
+        cell.backgroundColor = backgroundColor
+        cell.textLabel?.textColor = unselectedTextColor
+        cell.textLabel?.textAlignment = .center
+        setTableViewCellCornerRadius(cell: cell)
+        
+        // If the Floor Filter state is fully expanded then set the title of the cell based on the levels list
+        // Otherwise set the title of cell to the selected level short name
+        if (state == .fullyExpanded) {
+            if (!levels.isEmpty) {
+                cell.textLabel?.text = levels[indexPath.row].shortName
+            }
+        } else {
+            cell.textLabel?.text = viewModel.selectedLevel?.shortName
+                ?? viewModel.getDefaultLevelForFacility(facility: viewModel.selectedFacility)?.shortName
+        }
+        
+        let visibleLevelVerticalOrder = levels.first { $0.isVisible }.map { $0.verticalOrder }
+        let levelShortNames = levels.filter { $0.verticalOrder == visibleLevelVerticalOrder }.map {$0.shortName}
+        if (levelShortNames.contains(cell.textLabel?.text ?? "")) {
+            cell.backgroundColor = selectionColor
+            cell.textLabel?.textColor = selectedTextColor
+        }
+       
+        return cell
+    }
+    
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if (state == .fullyExpanded) {
+            viewModel.selectedLevel = viewModel.visibleLevelsInExpandedList[indexPath.row]
+            if let onSelectedLevelChangedListener = onSelectedLevelChangedListener {
+                onSelectedLevelChangedListener()
+            }
+        }
+        
+        // If the Floor Filter state is partially expanded, then expand the state fully
+        // Do not set the selected level
+        if (state == .partiallyExpanded) {
+            state = .fullyExpanded
+            updateViewsVisibilityForState(state: state)
+        }
+        
+        viewModel.filterMapToSelectedLevel()
+        levelsTableView.reloadData()
```

Comment:
That's a good observation. I made that change for the facilities and sites tableView. However, for this case with the levels table, when I tried to reload the previously selected and currently selected row, the UI wasn't sizing the cells properly. Additionally, there is a property `maxDisplayLevels` that controls the visible cells. Its default is 3, so it would not be that inefficient to load 3 visible cells since we would 2 cells each time already (the previously selected and currently selected).

Reply:
> when I tried to reload the previously selected and currently selected row, the UI wasn't sizing the cells properly.

I'm curious whether you tried calling `beginUpdates()` and `endUpdates()`? It may seem odd, but it is a documented way of causing the table view to lay out the existing cells without reloading them:

![image](https://user-images.githubusercontent.com/2257493/151418184-30ec188d-b6bd-4a11-8220-2098c06f8585.png)

I have had to do that in the past. `performBatchUpdates(_:completion:)` may do the same thing, but it isn't documented and I don't remember if I've tried it before.

Reply:
@philium I did try the `beginUpdates()` and `endUpdates()`. However, the cells are still not resizing properly. It takes 2 clicks to get the cells resize. I'd like to keep the `levelsTableView.reloadData()` for this initial implementation and focus on enhancing this logic in the future. Additionally, we expect users to show only 3 levels at a time in the tableView, so it is not that expensive to reload the table.